src/component.cpp

```diff
@@ -153,19 +149,21 @@ void Component::removeAllVariables()
 
 VariablePtr Component::getVariable(size_t index) const
 {
+    VariablePtr variable = nullptr;
     if (index < mPimpl->mVariables.size()) {
-        return mPimpl->mVariables.at(index);
+        variable = mPimpl->mVariables.at(index);
     }
-    return nullptr;
+    return variable;
```

Comment:
Why overcomplicate things by declaring and using `variable`? (Same below and elsewhere in the code.)